tests/shell/src/main.c

```diff
@@ -316,6 +317,32 @@ static void test_set_root_cmd(void)
 	test_shell_execute_cmd("shell colors on", 0);
 }
 
+static void test_shell_vfprintf(void)
+{
+	const char expect[] = "testing 1 2 3";
+	const struct shell *shell;
+	const char *buf, *start;
+	ssize_t size;
+
+	shell = shell_backend_record_get_ptr();
+	zassert_not_null(shell, "Failed to get shell");
+
+	buf = shell_backend_record_get_output(shell, &size);
+	shell_fprintf(shell, SHELL_VT100_COLOR_DEFAULT, "testing %d %s %c",
```

Comment:
This is a bit misleading as `test_shell_vfprintf` is not corresponding with `shell_fprintf`

Reply:
done